### PR TITLE
57 production manager perms

### LINKED ISSUE #57: Production manager can't download/edit/view history of files uploaded by the typesetter
The icons are not rendered
image

### LINKED ISSUE #77: Change how production managers download/delete galleys
The current flow to delete a galley at the moment is to edit a galley and then find the linked file. Hit delete and the plugin will delete the file and any associated galleys.
Instead, we could have the delete button on the galley record. The user clicks the delete button on the galley, and then we can decide whether or not to delete the file.

This has two main advantages:
 - The production manager doesn't have to go onto the edit menu before deleting a galley
 - At the implementation level, it is easier to reason about permissions on a galley object than it is about a file object.

All download links for editors/production staff should be update to use the typesetting download view.

### PR DESCRIPTION
**Pull request by ajrbyers**
There is room for improvement here but it will work better than core does currently.

- Adds can_manage_file security logic, decorator and templatetag
- Adds new Download file view
- Adds new Delete file/galley view

Closes #57 
Closes #77

### CODE REVIEW

File: views.py [hunk 1]
```diff
@@ -668,6 +668,63 @@ def typesetting_typesetter_download_file(request, assignment_id, file_id):
...
+    )
+
+    article = get_object_or_404(
+        submission_models.Article,
+        pk=file.article_id,
+    )
+
+
+    # Check how many galleys there are before deleting them.
+    galley_count = file.galley_set.all().count()
```
Comment (mauromsl) [inline views.py:708]: `file.delete()` returns a dict with a count of all the related objects that have been deleted, we don't need this count here

File: views.py [hunk 2]
```diff
@@ -668,6 +668,63 @@ def typesetting_typesetter_download_file(request, assignment_id, file_id):
...
+    file.galley_set.all()
```
Comment (mauromsl) [inline views.py:709]: This looks like a debug statement

File: views.py [hunk 3]
```diff
@@ -1103,6 +1160,9 @@ def typesetting_preview_galley(
         pk=article_id,
         journal=request.journal,
     )
+    allowed_roles = ['editor', 'section-editor', 'production']
+    print(request.user.has_role(request, allowed_roles))
```
Comment (mauromsl) [inline views.py:1164]: debug statement

File: security.py [hunk 4]
```diff
@@ -69,3 +69,46 @@ def inner(request, *args, **kwargs):
...
+    if request.user.is_anonymous():
+        return False
+
+    if (
+        request.user.is_staff or 
+        request.user.is_editor(request) or
+        request.user.is_production(request) or 
+        file_object.owner == request.user
+    ):
+        return True
```
Comment (mauromsl) [inline security.py:111]: We should at least check the file against the article and verify that there is a workflow element for the 'typesetting_plugin' stage recorded for it.

File: tests.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+from plugins.typesetting import plugin_settings
+from utils.testing import helpers
+
+
+class TestTypesetting(TestCase):
+    # Tests for editor role checks
+
+    @classmethod
+    def setUpTestData(self):
```
Comment (mauromsl) [inline tests.py:14]: The setup method looks ok, but there aren't any tests AFAIC

**Comment by ajrbyers** [comment]
Yeah - I was meant to turn this into a draft 'cause its not done yet 😭

File: views.py (hunk 2 above)
Comment (ajrbyers) [reply views.py:709]: Its actually meant to have `.delete()` after it woops.

File: tests.py (hunk 5 above)
Comment (ajrbyers) [reply tests.py:177]: Done.

File: security.py [hunk 6]
```diff
@@ -64,8 +64,64 @@ def inner(request, *args, **kwargs):
...
+    return wrapper
+
+
+
+def can_manage_file(request, file_object):
+    """
+    Determines if a user can view and download a file in the Typesetting Plugin.
+    """
+    if request.user.is_anonymous():
+        return False
```
Comment (mauromsl) [inline security.py:105]: Should we redirect to login like the `base_check` does in the security app?

**Comment by mauromsl** [comment]
Now it also depends on https://github.com/BirkbeckCTP/janeway/pull/1558

File: security.py [hunk 7]
```diff
@@ -64,8 +64,69 @@ def inner(request, *args, **kwargs):
...
+                reverse('core_login'),
+                request.path_info
+            )
+        )
+
+    if file_object.article_id:
+        # Check if there is a workflow log entry for the typesetting plugin.
+        if not core_models.WorkflowLog.objects.filter(
+            article__pk=file_object.article_id,
+            element__element_name=plugin_settings.PLUGIN_NAME,
```
Comment (mauromsl) [inline security.py:116]: worth checking `article__journal=request.journal` to avoid cross-journal URL forgery

File: views.py [hunk 8]
```diff
@@ -1252,3 +1253,23 @@ def preview_figure(
...
+def article_file_make_galley(request, article_id, file_id):
+    """ Copies a file to be a publicly available galley
+
+    :param request: the request associated with this call
+    :param article_id: the ID of the associated articled
+    :param file_id: the file ID for which to view the history
+    :return: a redirect to the URL at the GET parameter 'return'
+    """
+    article_object = get_object_or_404(
+        submission_models.Article.allarticles, pk=article_id)
```
Comment (mauromsl) [inline views.py:1268]: Should be `submission_models.Article`

**Comment by ajrbyers** [comment]
The PR that will never die!

**Pull request merged**